### PR TITLE
Login screen fix

### LINKED ISSUE #21702: Implement error handling for manual Add Login screen
Settings -> Logins and passwords -> Saved logins -> Add login

Confirm the following error handling exists/implement it: 

**Hostname**
* must contain "https://" or "http://"
* may not be blank
* must be a valid url (can't be something like "http://gggg")
* may not be blank

**Username**
* N/A
* **Can** be blank

**Password**
* May not be blank

image

**Notes**
* When the Add login screen is first shown and all fields are empty, there should be no red error text
* Once the user begins typing/navigates between fields, error text should be shown
* The save button should not be enabled if there are any errors

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-18170)

### PR DESCRIPTION
**Pull request by joaopmatos**
Fixes #21702 
### Pull Request checklist
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

Upon entering the login information screen, both the hostname and password are required upfront. The username can be left blank, although if it is edited, it becomes a required field.
requiredInfo
Once we begin adding the hostname, it's required that the hostame follows the usual structure of scheme https:// or https://, subdomain and top-level domain.
requiredHostname
Once the hostname is considered valid, the user still needs to insert a password for the login to be considered valid and the "confirm" button to become available to click.
requiredBoth
Upon entering a valid password as well, all is well and the login can be confirmed.
requiredPassword

On various attempts of finding additional acessibility difficulties that could have come up, none extra were found. However, I never managed to have an identical number of suggestions on both Firefox Preview and Firefox on my emulator.

I have had mutiple attempts at bypassing the correct parameters for the login, using random hostname combinations, including different schemes, i.e ftp://. In every single attempt I've made the "confirm" button only was available upon having both a valid hostname and password, additionally, a valid username if the user desired to type one.

Summing up the changes: now the hostname and password are always required, and the username is only required if the user attempts to type one. Upon having a valid hostname and password, the user may proceed.

If any change is needed please let me know as this will be my first contribution!

### CODE REVIEW

**Comment by eliserichards** [comment]
Thank you so much for taking this on @joaopmatos! I will pull down your branch and do some testing 👍

**Review by eliserichards — CHANGES_REQUESTED** [review]
Looks like there are some linting errors @joaopmatos. Can you fix them and then run `./gradlew ktlint` to confirm they are fixed? --

```
/home/runner/work/fenix/fenix/app/src/main/java/org/mozilla/fenix/settings/logins/fragment/AddLoginFragment.kt:19:1: Unused import
/home/runner/work/fenix/fenix/app/src/main/java/org/mozilla/fenix/settings/logins/fragment/AddLoginFragment.kt:170:71: Unnecessary space(s)
```

**Comment by joaopmatos** [comment]
Hey @eliserichards, thanks for the reply! I've one the corrections you pointed out, but now I get this error running _./gradlew ktlint_: 

* Where:
Build file '/home/joao/AndroidStudioProjects/fenix/app/build.gradle' line: 90

* What went w
rong:
A problem occurred evaluating project ':app'.
> command exited with non-zero exit value: 129.
Screenshot_20211128_204412


Can you help me out? I've already searched quite a bit without finding a solution. I've tried even to clean the project and rebuild it, but to no success.

**Comment by boek** [comment]
This was fixed #26736
